### ISSUE TITLE
Feature Request: Add a way to prompt for and read keys from terminal without need for pressing Enter (raw input).

### ISSUE THREAD

**Issue by Animeshz** [ajalt/clikt#279]
**Motivation/UseCases:**

1. Can be used to input `y` or `n` without enter.

```bash
Do you want to download this file? (y/n): y
Downloading [----            ]
```

2. Can be used to make mini-games in cli.

3. Can be used for making interaction with modern terminals like [fzf](https://github.com/junegunn/fzf#-) (for searching as soon as query is entered) or [interact](https://github.com/frencojobs/interact#overview) (for making something like options and navigating via arrow keys).

**Comment by Animeshz** [comment]
This would require to turn on the raw mode in the terminals, I've found some resources regarding it:

1. For Windows: https://github.com/timsneath/dart_console/blob/main/lib/src/ffi/win/termlib-win.dart
2. For Unix-like OS: https://github.com/timsneath/dart_console/blob/main/lib/src/ffi/unix/termlib-unix.dart

The implementation would require some native (c/c++) calls. That can either be implemented by jni/napi in jvm and js respectively and k/n has c-interop built in, I also have proof of concept in one of [my project](https://github.com/Animeshz/keyboard-mouse-kt), but that could be quite cumbersome for a single feature request, as native libs need to be compiled for each platform even if they have the same content.

The second way is through a ffi (foreign function interface) which is available in both jvm and js (node) which is also used in that project I linked (of dart). Which should be perfect fit as I/O bounded tasks already are slow, we might not get much benefit of implementation via napi/jni, and we'll get it os/architecture independent (don't need to compile native things).
